File: Financial Fraud Detection using Predictive Analytics/scripts/r/statistical_analysis.R
# Statistical Analysis Script for Fraud Detection Dataset (R)
#
# This script performs comprehensive statistical analysis including:
# 1. Descriptive Statistics - Mean, median, mode, standard deviation, variance, skewness, kurtosis
# 2. Inferential Statistics - Hypothesis testing, confidence intervals, t-tests, chi-square tests
# 3. Exploratory Statistics - Correlation analysis, feature relationships, statistical tests

# Load libraries
library(tidyverse)
library(car)
library(psych)
library(corrplot)
library(ggplot2)

# Set options
options(warn = -1)
set.seed(42)

# Set up paths
project_root <- dirname(dirname(dirname(getwd())))
data_path <- file.path(project_root, "data", "fraud_data.csv")
output_dir <- file.path(project_root, "outputs", "figures")

# Create output directory if it doesn't exist
dir.create(output_dir, recursive = TRUE, showWarnings = FALSE)

# Load data
cat("Loading data...\n")
df <- read.csv(data_path, stringsAsFactors = FALSE)
cat("Data loaded:", dim(df), "\n")
cat("Target variable distribution:\n")
print(table(df$isFraud))

# ============================================================================
# 1. DESCRIPTIVE STATISTICS
# ============================================================================
cat("\n", rep("=", 80), "\n", sep = "")
cat("DESCRIPTIVE STATISTICS\n")
cat(rep("=", 80), "\n", sep = "")

# Key features for analysis
key_features <- c("TransactionAmt", "card1", "card2", "card3", "card5")
key_features <- key_features[key_features %in% colnames(df)]

if(length(key_features) > 0) {
  # Descriptive statistics
  desc_stats <- psych::describe(df[key_features])
  cat("\nDescriptive Statistics for Key Numerical Features:\n")
  print(desc_stats)
  
  # Statistics by fraud status
  if("TransactionAmt" %in% key_features) {
    cat("\nTransaction Amount Statistics by Fraud Status:\n")
    fraud_stats <- df %>%
      group_by(isFraud) %>%
      summarise(
        Count = n(),
        Mean = mean(TransactionAmt, na.rm = TRUE),
        Median = median(TransactionAmt, na.rm = TRUE),
        Std = sd(TransactionAmt, na.rm = TRUE),
        Min = min(TransactionAmt, na.rm = TRUE),
        Max = max(TransactionAmt, na.rm = TRUE),
        Skewness = psych::skew(TransactionAmt, na.rm = TRUE),
        Kurtosis = psych::kurtosi(TransactionAmt, na.rm = TRUE),
        .groups = 'drop'
      )
    print(fraud_stats)
    
    # Visualization
    png(file.path(output_dir, "descriptive_stats_transaction_r.png"), 
        width = 1200, height = 600, res = 300)
    par(mfrow = c(1, 2))
    
    # Histogram
    hist(df$TransactionAmt[df$isFraud == 0], breaks = 50, col = rgb(0, 0, 1, 0.5),
         main = "Transaction Amount Distribution by Fraud Status",
         xlab = "Transaction Amount", ylab = "Frequency", xlim = c(0, quantile(df$TransactionAmt, 0.99, na.rm = TRUE)))
    hist(df$TransactionAmt[df$isFraud == 1], breaks = 50, col = rgb(1, 0, 0, 0.5), add = TRUE)
    legend("topright", legend = c("Legitimate", "Fraud"), fill = c(rgb(0, 0, 1, 0.5), rgb(1, 0, 0, 0.5)))
    
    # Box plot
    boxplot(TransactionAmt ~ isFraud, data = df, log = "y",
            main = "Transaction Amount by Fraud Status",
            xlab = "Fraud Status", ylab = "Transaction Amount (log scale)",
            names = c("Legitimate", "Fraud"))
    
    dev.off()
    cat("\nFigure saved to", file.path(output_dir, "descriptive_stats_transaction_r.png"), "\n")
  }
}

# ============================================================================
# 2. INFERENTIAL STATISTICS
# ============================================================================
cat("\n", rep("=", 80), "\n", sep = "")
cat("INFERENTIAL STATISTICS\n")
cat(rep("=", 80), "\n", sep = "")

# Hypothesis Test 1: Transaction Amount
if("TransactionAmt" %in% colnames(df)) {
  cat("\nHypothesis Test: Transaction Amount\n")
  cat("H0: Mean transaction amount is the same for fraud and legitimate transactions\n")
  cat("H1: Mean transaction amount differs between fraud and legitimate transactions\n\n")
  
  fraud_amt <- df$TransactionAmt[df$isFraud == 1 & !is.na(df$TransactionAmt)]
  legit_amt <- df$TransactionAmt[df$isFraud == 0 & !is.na(df$TransactionAmt)]
  
  # Mann-Whitney U test (non-parametric)
  test_result <- wilcox.test(fraud_amt, legit_amt, alternative = "two.sided")
  cat("Mann-Whitney U Test Results:\n")
  print(test_result)
  
  if(test_result$p.value < 0.05) {
    cat("Result: REJECT H0 - Mean transaction amounts are significantly different\n")
  } else {
    cat("Result: FAIL TO REJECT H0 - No significant difference in mean transaction amounts\n")
  }
  
  # Confidence intervals
  fraud_ci <- t.test(fraud_amt)$conf.int
  legit_ci <- t.test(legit_amt)$conf.int
  
  cat("\n95% Confidence Intervals:\n")
  cat("Fraud transactions:", fraud_ci[1], "-", fraud_ci[2], 
      "(mean:", mean(fraud_amt), ")\n")
  cat("Legitimate transactions:", legit_ci[1], "-", legit_ci[2], 
      "(mean:", mean(legit_amt), ")\n")
}

# Hypothesis Test 2: Chi-square test for ProductCD
if("ProductCD" %in% colnames(df)) {
  cat("\n", rep("=", 80), "\n", sep = "")
  cat("Chi-square Test: ProductCD and Fraud\n")
  cat(rep("=", 80), "\n", sep = "")
  
  contingency_table <- table(df$ProductCD, df$isFraud)
  cat("\nContingency Table:\n")
  print(contingency_table)
  
  chi_test <- chisq.test(contingency_table)
  cat("\nChi-square Test Results:\n")
  print(chi_test)
  
  if(chi_test$p.value < 0.05) {
    cat("Result: REJECT H0 - ProductCD is significantly associated with fraud\n")
  } else {
    cat("Result: FAIL TO REJECT H0 - No significant association\n")
  }
  
  # Visualization
  contingency_pct <- prop.table(contingency_table, margin = 1) * 100
  png(file.path(output_dir, "chi_square_productcd_r.png"), 
      width = 1200, height = 800, res = 300)
  barplot(t(contingency_pct), main = "Fraud Rate by ProductCD", 
          xlab = "ProductCD", ylab = "Percentage (%)",
          legend.text = c("Legitimate", "Fraud"), 
          args.legend = list(x = "topright"))
  dev.off()
  cat("\nFigure saved to", file.path(output_dir, "chi_square_productcd_r.png"), "\n")
}

# Hypothesis Test 3: Chi-square test for card type
if("card4" %in% colnames(df)) {
  cat("\n", rep("=", 80), "\n", sep = "")
  cat("Chi-square Test: Card Type (card4) and Fraud\n")
  cat(rep("=", 80), "\n", sep = "")
  
  contingency_table <- table(df$card4, df$isFraud)
  cat("\nContingency Table:\n")
  print(contingency_table)
  
  chi_test <- chisq.test(contingency_table)
  cat("\nChi-square Test Results:\n")
  print(chi_test)
  
  if(chi_test$p.value < 0.05) {
    cat("Result: REJECT H0 - Card type is significantly associated with fraud\n")
  } else {
    cat("Result: FAIL TO REJECT H0 - No significant association\n")
  }
}

# ============================================================================
# 3. EXPLORATORY STATISTICS
# ============================================================================
cat("\n", rep("=", 80), "\n", sep = "")
cat("EXPLORATORY STATISTICS\n")
cat(rep("=", 80), "\n", sep = "")

# Correlation analysis
cat("\nCorrelation Analysis\n")
key_features <- c("TransactionAmt", "card1", "card2", "card3", "card5", "isFraud")
key_features <- key_features[key_features %in% colnames(df)]

if(length(key_features) > 1) {
  # Calculate correlation matrix
  corr_matrix <- cor(df[key_features], use = "complete.obs")
  
  # Visualization
  png(file.path(output_dir, "correlation_matrix_statistical_r.png"), 
      width = 1200, height = 1000, res = 300)
  corrplot(corr_matrix, method = "color", type = "upper", 
           order = "hclust", tl.cex = 0.8, tl.col = "black",
           addCoef.col = "black", number.cex = 0.7)
  dev.off()
  cat("\nFigure saved to", file.path(output_dir, "correlation_matrix_statistical_r.png"), "\n")
  
  # Correlation with fraud
  if("isFraud" %in% key_features) {
    fraud_corr <- corr_matrix[, "isFraud"]
    fraud_corr <- sort(fraud_corr, decreasing = TRUE)
    cat("\nFeatures most correlated with Fraud:\n")
    print(fraud_corr)
  }
}

# Statistical summary by fraud status
cat("\n", rep("=", 80), "\n", sep = "")
cat("Statistical Summary by Fraud Status\n")
cat(rep("=", 80), "\n", sep = "")

analysis_features <- c("TransactionAmt", "card1", "card2", "card3", "card5")
analysis_features <- analysis_features[analysis_features %in% colnames(df)]

if(length(analysis_features) > 0) {
  summary_stats <- df %>%
    group_by(isFraud) %>%
    summarise(across(all_of(analysis_features), 
                     list(mean = ~mean(.x, na.rm = TRUE),
                          median = ~median(.x, na.rm = TRUE),
                          sd = ~sd(.x, na.rm = TRUE)),
                     .names = "{.col}_{.fn}"))
  cat("\nSummary Statistics by Fraud Status:\n")
  print(summary_stats)
  
  # Perform statistical tests for each feature
  cat("\nStatistical Tests for Each Feature:\n")
  test_results <- data.frame(
    Feature = character(),
    Test = character(),
    Statistic = numeric(),
    p.value = numeric(),
    Significant = character(),
    stringsAsFactors = FALSE
  )
  
  for(feature in analysis_features) {
    fraud_data <- df[[feature]][df$isFraud == 1 & !is.na(df[[feature]])]
    legit_data <- df[[feature]][df$isFraud == 0 & !is.na(df[[feature]])]
    
    if(length(fraud_data) > 0 && length(legit_data) > 0) {
      # Mann-Whitney U test
      test_result <- wilcox.test(fraud_data, legit_data, alternative = "two.sided")
      test_results <- rbind(test_results, data.frame(
        Feature = feature,
        Test = "Mann-Whitney U",
        Statistic = test_result$statistic,
        p.value = test_result$p.value,
        Significant = ifelse(test_result$p.value < 0.05, "Yes", "No"),
        stringsAsFactors = FALSE
      ))
    }
  }
  
  print(test_results)
}

# Summary
cat("\n", rep("=", 80), "\n", sep = "")
cat("STATISTICAL ANALYSIS SUMMARY\n")
cat(rep("=", 80), "\n", sep = "")
cat("\n1. Descriptive Statistics:\n")
cat("   - Calculated mean, median, std, skewness, and kurtosis for key features\n")
cat("   - Compared statistics between fraud and legitimate transactions\n")
cat("\n2. Inferential Statistics:\n")
cat("   - Performed hypothesis tests to compare fraud and legitimate transactions\n")
cat("   - Used non-parametric tests (Mann-Whitney U) due to non-normal distributions\n")
cat("   - Calculated confidence intervals for key metrics\n")
cat("   - Performed chi-square tests for categorical variables\n")
cat("\n3. Exploratory Statistics:\n")
cat("   - Analyzed correlations between features and fraud\n")
cat("   - Identified statistically significant relationships\n")
cat("   - Explored feature distributions by fraud status\n")
cat("\nStatistical Analysis Complete! Check outputs/figures/ for visualizations.\n")
cat(rep("=", 80), "\n", sep = "")

File: Financial Fraud Detection using Predictive Analytics/scripts/python/statistical_analysis.py
"""
Statistical Analysis Script for Fraud Detection Dataset

This script performs comprehensive statistical analysis including:
1. Descriptive Statistics - Mean, median, mode, standard deviation, variance, skewness, kurtosis
2. Inferential Statistics - Hypothesis testing, confidence intervals, t-tests, chi-square tests
3. Exploratory Statistics - Correlation analysis, feature relationships, statistical tests
"""

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
from scipy.stats import normaltest, shapiro, chi2_contingency, ttest_ind, mannwhitneyu
import warnings
from pathlib import Path
from statsmodels.stats.proportion import proportions_ztest

warnings.filterwarnings('ignore')
sns.set_style("whitegrid")
np.random.seed(42)

def descriptive_statistics(df, output_dir):
    """Perform descriptive statistical analysis."""
    print("\n" + "="*80)
    print("DESCRIPTIVE STATISTICS")
    print("="*80)
    
    numerical_cols = df.select_dtypes(include=[np.number]).columns.tolist()
    key_features = ['TransactionAmt', 'card1', 'card2', 'card3', 'card5', 'addr1', 'addr2']
    key_features = [f for f in key_features if f in numerical_cols]
    
    print("\nDescriptive Statistics for Key Numerical Features:")
    desc_stats = df[key_features].describe()
    print(desc_stats)
    
    # Additional statistics: skewness and kurtosis
    print("\nSkewness and Kurtosis:")
    skew_kurt = pd.DataFrame({
        'Skewness': df[key_features].skew(),
        'Kurtosis': df[key_features].kurtosis()
    })
    print(skew_kurt)
    
    # Descriptive statistics by fraud status
    if 'TransactionAmt' in df.columns:
        print("\nTransaction Amount Statistics by Fraud Status:")
        fraud_stats = df.groupby('isFraud')['TransactionAmt'].agg([
            'count', 'mean', 'median', 'std', 'min', 'max', 
            'skew', pd.Series.kurt
        ]).round(4)
        fraud_stats.columns = ['Count', 'Mean', 'Median', 'Std', 'Min', 'Max', 'Skewness', 'Kurtosis']
        print(fraud_stats)
        
        # Visualize distributions
        fig, axes = plt.subplots(1, 2, figsize=(14, 5))
        
        # Histogram
        df[df['isFraud']==0]['TransactionAmt'].hist(bins=50, alpha=0.7, label='Legitimate', ax=axes[0], color='blue')
        df[df['isFraud']==1]['TransactionAmt'].hist(bins=50, alpha=0.7, label='Fraud', ax=axes[0], color='red')
        axes[0].set_xlabel('Transaction Amount')
        axes[0].set_ylabel('Frequency')
        axes[0].set_title('Transaction Amount Distribution by Fraud Status')
        axes[0].legend()
        axes[0].set_xlim(0, df['TransactionAmt'].quantile(0.99))
        
        # Box plot
        df.boxplot(column='TransactionAmt', by='isFraud', ax=axes[1])
        axes[1].set_xlabel('Fraud Status')
        axes[1].set_ylabel('Transaction Amount')
        axes[1].set_title('Transaction Amount by Fraud Status')
        axes[1].set_yscale('log')
        plt.suptitle('')
        plt.tight_layout()
        plt.savefig(f'{output_dir}/descriptive_stats_transaction.png', dpi=300, bbox_inches='tight')
        plt.close()
        print(f"\nFigure saved to {output_dir}/descriptive_stats_transaction.png")

def inferential_statistics(df, output_dir):
    """Perform inferential statistical analysis."""
    print("\n" + "="*80)
    print("INFERENTIAL STATISTICS")
    print("="*80)
    
    # Hypothesis Test 1: Test if mean transaction amount differs between fraud and legitimate transactions
    if 'TransactionAmt' in df.columns:
        print("\nHypothesis Test: Transaction Amount")
        print("H0: Mean transaction amount is the same for fraud and legitimate transactions")
        print("H1: Mean transaction amount differs between fraud and legitimate transactions")
        print("\n")
        
        fraud_amt = df[df['isFraud']==1]['TransactionAmt'].dropna()
        legit_amt = df[df['isFraud']==0]['TransactionAmt'].dropna()
        
        # Check normality (Shapiro-Wilk test on sample)
        sample_size = min(5000, len(fraud_amt), len(legit_amt))
        _, p_fraud_norm = shapiro(fraud_amt.sample(sample_size, random_state=42))
        _, p_legit_norm = shapiro(legit_amt.sample(sample_size, random_state=42))
        
        print(f"Normality test p-values:")
        print(f"  Fraud transactions: {p_fraud_norm:.6f}")
        print(f"  Legitimate transactions: {p_legit_norm:.6f}")
        print(f"  (Both distributions are likely non-normal, p < 0.05)")
        
        # Use Mann-Whitney U test (non-parametric)
        statistic, p_value = mannwhitneyu(fraud_amt, legit_amt, alternative='two-sided')
        print(f"\nMann-Whitney U Test Results:")
        print(f"  Statistic: {statistic:.4f}")
        print(f"  p-value: {p_value:.6f}")
        print(f"  Significance level: 0.05")
        
        if p_value < 0.05:
            print(f"  Result: REJECT H0 - Mean transaction amounts are significantly different")
        else:
            print(f"  Result: FAIL TO REJECT H0 - No significant difference in mean transaction amounts")
        
        # Also perform t-test for comparison
        t_stat, t_pvalue = ttest_ind(fraud_amt, legit_amt)
        print(f"\nIndependent t-test Results (for comparison):")
        print(f"  t-statistic: {t_stat:.4f}")
        print(f"  p-value: {t_pvalue:.6f}")
        
        # Calculate confidence intervals
        fraud_mean = fraud_amt.mean()
        fraud_std = fraud_amt.std()
        fraud_se = fraud_std / np.sqrt(len(fraud_amt))
        fraud_ci = stats.norm.interval(0.95, loc=fraud_mean, scale=fraud_se)
        
        legit_mean = legit_amt.mean()
        legit_std = legit_amt.std()
        legit_se = legit_std / np.sqrt(len(legit_amt))
        legit_ci = stats.norm.interval(0.95, loc=legit_mean, scale=legit_se)
        
        print(f"\n95% Confidence Intervals:")
        print(f"  Fraud transactions: ${fraud_ci[0]:.2f} - ${fraud_ci[1]:.2f} (mean: ${fraud_mean:.2f})")
        print(f"  Legitimate transactions: ${legit_ci[0]:.2f} - ${legit_ci[1]:.2f} (mean: ${legit_mean:.2f})")
    
    # Hypothesis Test 2: Chi-square test for categorical variables
    print("\n" + "="*80)
    print("Chi-square Test: ProductCD and Fraud")
    print("="*80)
    
    if 'ProductCD' in df.columns:
        # Create contingency table
        contingency_table = pd.crosstab(df['ProductCD'], df['isFraud'])
        print("\nContingency Table:")
        print(contingency_table)
        
        # Chi-square test
        chi2, p_value, dof, expected = chi2_contingency(contingency_table)
        
        print(f"\nChi-square Test Results:")
        print(f"  Chi-square statistic: {chi2:.4f}")
        print(f"  p-value: {p_value:.6f}")
        print(f"  Degrees of freedom: {dof}")
        print(f"  Significance level: 0.05")
        
        if p_value < 0.05:
            print(f"  Result: REJECT H0 - ProductCD is significantly associated with fraud")
        else:
            print(f"  Result: FAIL TO REJECT H0 - No significant association")
        
        # Visualize
        contingency_pct = contingency_table.div(contingency_table.sum(axis=1), axis=0) * 100
        contingency_pct.plot(kind='bar', stacked=True, figsize=(10, 6))
        plt.title('Fraud Rate by ProductCD', fontsize=14, fontweight='bold')
        plt.xlabel('ProductCD')
        plt.ylabel('Percentage (%)')
        plt.legend(['Legitimate', 'Fraud'])
        plt.xticks(rotation=0)
        plt.tight_layout()
        plt.savefig(f'{output_dir}/chi_square_productcd.png', dpi=300, bbox_inches='tight')
        plt.close()
        print(f"\nFigure saved to {output_dir}/chi_square_productcd.png")
    
    # Hypothesis Test 3: Test for card type and fraud
    if 'card4' in df.columns:
        print("\n" + "="*80)
        print("Chi-square Test: Card Type (card4) and Fraud")
        print("="*80)
        
        contingency_table = pd.crosstab(df['card4'], df['isFraud'])
        print("\nContingency Table:")
        print(contingency_table)
        
        # Chi-square test
        chi2, p_value, dof, expected = chi2_contingency(contingency_table)
        
        print(f"\nChi-square Test Results:")
        print(f"  Chi-square statistic: {chi2:.4f}")
        print(f"  p-value: {p_value:.6f}")
        print(f"  Degrees of freedom: {dof}")
        
        if p_value < 0.05:
            print(f"  Result: REJECT H0 - Card type is significantly associated with fraud")
        else:
            print(f"  Result: FAIL TO REJECT H0 - No significant association")

def exploratory_statistics(df, output_dir):
    """Perform exploratory statistical analysis."""
    print("\n" + "="*80)
    print("EXPLORATORY STATISTICS")
    print("="*80)
    
    # Correlation analysis
    print("\nCorrelation Analysis")
    key_features = ['TransactionAmt', 'card1', 'card2', 'card3', 'card5', 
                    'addr1', 'addr2', 'dist1', 'dist2', 'isFraud']
    key_features = [f for f in key_features if f in df.columns and df[f].dtype in [np.int64, np.float64]]
    
    if len(key_features) > 1:
        # Calculate correlation matrix
        corr_matrix = df[key_features].corr()
        
        # Visualize correlation matrix
        plt.figure(figsize=(12, 10))
        sns.heatmap(corr_matrix, annot=True, fmt='.3f', cmap='coolwarm', center=0,
                    square=True, linewidths=1, cbar_kws={"shrink": 0.8})
        plt.title('Correlation Matrix of Key Features', fontsize=14, fontweight='bold')
        plt.tight_layout()
        plt.savefig(f'{output_dir}/correlation_matrix_statistical.png', dpi=300, bbox_inches='tight')
        plt.close()
        print(f"\nFigure saved to {output_dir}/correlation_matrix_statistical.png")
        
        # Features most correlated with fraud
        if 'isFraud' in corr_matrix.columns:
            fraud_corr = corr_matrix['isFraud'].sort_values(ascending=False)
            print("\nFeatures most correlated with Fraud:")
            print(fraud_corr)
            
            # Statistical significance of correlations
            print("\nStatistical Significance of Correlations with Fraud:")
            n = len(df)
            significant_features = []
            for feature in fraud_corr.index:
                if feature != 'isFraud':
                    corr_coef = fraud_corr[feature]
                    # Test significance of correlation
                    t_stat = corr_coef * np.sqrt((n - 2) / (1 - corr_coef**2))
                    p_value = 2 * (1 - stats.t.cdf(abs(t_stat), n - 2))
                    if p_value < 0.05:
                        significant_features.append((feature, corr_coef, p_value))
            
            if significant_features:
                sig_df = pd.DataFrame(significant_features, columns=['Feature', 'Correlation', 'p-value'])
                print("\nSignificantly correlated features (p < 0.05):")
                print(sig_df)
    
    # Statistical summary by fraud status for multiple features
    print("\n" + "="*80)
    print("Statistical Summary by Fraud Status")
    print("="*80)
    
    analysis_features = ['TransactionAmt', 'card1', 'card2', 'card3', 'card5']
    analysis_features = [f for f in analysis_features if f in df.columns]
    
    if analysis_features:
        summary_stats = df.groupby('isFraud')[analysis_features].agg(['mean', 'median', 'std', 'skew'])
        print("\nSummary Statistics by Fraud Status:")
        print(summary_stats)
        
        # Perform statistical tests for each feature
        print("\nStatistical Tests for Each Feature:")
        test_results = []
        
        for feature in analysis_features:
            fraud_data = df[df['isFraud']==1][feature].dropna()
            legit_data = df[df['isFraud']==0][feature].dropna()
            
            if len(fraud_data) > 0 and len(legit_data) > 0:
                # Mann-Whitney U test
                statistic, p_value = mannwhitneyu(fraud_data, legit_data, alternative='two-sided')
                test_results.append({
                    'Feature': feature,
                    'Test': 'Mann-Whitney U',
                    'Statistic': statistic,
                    'p-value': p_value,
                    'Significant': 'Yes' if p_value < 0.05 else 'No'
                })
        
        test_df = pd.DataFrame(test_results)
        print(test_df)

def main():
    """Main function to run statistical analysis."""
    # Set up paths
    project_root = Path(__file__).parent.parent.parent
    data_path = project_root / 'data' / 'fraud_data.csv'
    output_dir = project_root / 'outputs' / 'figures'
    output_dir.mkdir(parents=True, exist_ok=True)
    
    # Load data
    print("Loading data...")
    df = pd.read_csv(data_path)
    print(f"Data loaded: {df.shape}")
    print(f"Target variable distribution:\n{df['isFraud'].value_counts()}")
    
    # Perform analyses
    descriptive_statistics(df, output_dir)
    inferential_statistics(df, output_dir)
    exploratory_statistics(df, output_dir)
    
    # Summary
    print("\n" + "="*80)
    print("STATISTICAL ANALYSIS SUMMARY")
    print("="*80)
    print("\n1. Descriptive Statistics:")
    print("   - Calculated mean, median, std, skewness, and kurtosis for key features")
    print("   - Compared statistics between fraud and legitimate transactions")
    print("\n2. Inferential Statistics:")
    print("   - Performed hypothesis tests to compare fraud and legitimate transactions")
    print("   - Used non-parametric tests (Mann-Whitney U) due to non-normal distributions")
    print("   - Calculated confidence intervals for key metrics")
    print("   - Performed chi-square tests for categorical variables")
    print("\n3. Exploratory Statistics:")
    print("   - Analyzed correlations between features and fraud")
    print("   - Identified statistically significant relationships")
    print("   - Explored feature distributions by fraud status")
    print("\nStatistical Analysis Complete! Check outputs/figures/ for visualizations.")

if __name__ == "__main__":
    main()

